hw5/task2/script.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+document.addEventListener('DOMContentLoaded', () => {
+    const body = document.body;
+    const mainButton = document.querySelector('.main-btn');
+    const message = document.querySelector('.message p');
+    
+    const initTheme = localStorage.getItem('theme');
+
+    if (initTheme === 'light') {
+        lightTheme();
+    } else if (initTheme === 'dark') {
+        darkTheme();
+    }
+
+    mainButton.addEventListener('click', () => {
+        if (localStorage.getItem('theme') === 'dark') {
+            lightTheme(true);
+        } else {
+            darkTheme(true);
+        }
+    });
+
+    function lightTheme(changeLocalStorage = false) {
+        if (changeLocalStorage) {
+            localStorage.setItem('theme', 'light');
+            localStorage.setItem('themeChangeDate', formatDate(new Date()));
+        }
+        body.classList.remove('dark');
+        mainButton.textContent = 'Turn off';
+        message.innerText = `Last turn on: ${localStorage.getItem('themeChangeDate')}`;
+    }
+
+    function darkTheme(changeLocalStorage = false) {
+        if (changeLocalStorage) {
+            localStorage.setItem('theme', 'dark');
+            localStorage.setItem('themeChangeDate', formatDate(new Date()));
+        }
+        body.classList.add('dark');
+        mainButton.textContent = 'Turn on';
+        message.innerText = `Last turn off: ${localStorage.getItem('themeChangeDate')}`;
+    }
+
+    function formatDate(dateObj) {
+        return dateObj.toLocaleString(undefined, {day: '2-digit'})
+            + '-' + dateObj.toLocaleString(undefined, {month: '2-digit'})
+            + '-' + dateObj.toLocaleString(undefined, {year: 'numeric'})
+            + ' ' + dateObj.toLocaleString(undefined, {
+                hourCycle: 'h23',
+                hour: '2-digit',
+                minute: '2-digit',
+                second: '2-digit',
+            });
```

Comment:
Метод `toLocaleString` 

> returns a string with a language-sensitive representation of this date

Ти ж використовуєш його, без того, для чого він власне створений.

Для отримання необхідних значень з дати, треба використовувати відповідні методи.

Тому весь цей код можна зробити наступним чином.

```suggestion
        const day = dateObj.getDay();
        const month = dateObj.getMonth();
        const year = dateObj.getFullYear();
        const hour = dateObj.getHours();
        const minute = dateObj.getMinutes();
        const second = dateObj.getSeconds();

        return `${day}-${month}-${year} ${hour}:${minute}:${second}`;
```

Reply:
Цей код не робить абсолютно те саме, що і мій.
Має бути не <code>dateObj.getDay();</code> а <code>dateObj.getDate();</code> 
Та <code>month = dateObj.getMonth() + 1;</code> бо відлік починається з нуля.

А як зробити, щоб години, хвилини та секунди були з двох цифр?

Reply:
А мені й не стоїть задача написати за тебе робоче рішення)

Мені треба вказати на очевидну проблеми, і підштовхнути в напрямку її рішення.

> А як зробити, щоб години, хвилини та секунди були з двох цифр?

А ти пробувала запустити мій код? Там власне години, хвилини та секунди з двох цифр.